front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/PluginControllerSpec.groovy

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.controllers
+
+import com.netflix.spinnaker.front50.model.plugin.PluginRepository
+import com.netflix.spinnaker.front50.model.plugin.PluginRepository.GetCriteria
+import com.netflix.spinnaker.front50.model.plugin.PluginRepository.ListCriteria
+import com.netflix.spinnaker.front50.proto.Plugin
+import com.netflix.spinnaker.front50.proto.Plugins
+import spock.lang.Specification
+import spock.lang.Subject
+
+
+class PluginControllerSpec extends Specification {
+
+  def pluginBuilder = Plugin.newBuilder()
+    .setName("name")
+    .setNamespace("namespace")
+    .setVersion("1.0.0")
+    .setAuthor("author")
+    .addAllCapabilities(["a-capability", "another-capability"])
+    .addAllRequiredVersions(["clouddriver>=1", "orca>=2"])
+    .addAllDependencies(["a-necessary-thing"])
+    .setEnabled(true)
+    .setInProcessUnsafe(false)
+  Plugin plugin = pluginBuilder.build()
+  PluginRepository<Plugin> pluginRepository = Stub(PluginRepository)
+
+  @Subject
+  def controller = new PluginController(pluginRepository)
+
```

Comment:
Probably , having some MockMVC type tests here might be helpful to exercise 404/400 type scenarios and also for the happy path that deal with protobuf msgs ?